crates/oxc_linter/src/rules/jest/prefer_to_have_length.rs

```diff
@@ -138,52 +118,30 @@ impl PreferToHaveLength {
         let Some(matcher) = parsed_expect_call.matcher() else {
             return;
         };
-
         if expect_property_name != "length" || !is_equality_matcher(matcher) {
             return;
         }
 
         ctx.diagnostic_with_fix(use_to_have_length(matcher.span), |fixer| {
-            let code = Self::build_code(fixer, static_mem_expr, kind, property_name);
-            let end = if call_expr.arguments.len() > 0 {
-                call_expr.arguments.first().unwrap().span().start
-            } else {
-                matcher.span.end
+            let matcher_span = match fixer.source_range(matcher.span).chars().next().unwrap() {
+                '\'' | '"' | '`' => Span::new(matcher.span.start + 1, matcher.span.end - 1),
+                _ => matcher.span,
             };
-            fixer.replace(Span::new(call_expr.span.start, end - 1), code)
-        });
-    }
 
-    fn build_code<'a>(
-        fixer: RuleFixer<'_, 'a>,
-        mem_expr: &MemberExpression<'a>,
-        kind: Option<&str>,
-        property_name: Option<&str>,
-    ) -> String {
-        let mut formatter = fixer.codegen();
-        let Expression::Identifier(prop_ident) = mem_expr.object() else {
-            return formatter.into_source_text();
-        };
-
-        formatter.print_str("expect(");
-        formatter.print_str(prop_ident.name.as_str());
-        formatter.print_str(")");
-
-        if let Some(kind_val) = kind {
-            if kind_val == "ComputedMember" {
-                let property = property_name.unwrap();
-                formatter.print_str("[\"");
-                formatter.print_str(property);
-                formatter.print_str("\"]");
-            } else if kind_val == "StaticMember" {
-                formatter.print_str(".");
-                let property = property_name.unwrap();
-                formatter.print_str(property);
-            }
-        }
-
-        formatter.print_str(".toHaveLength");
-        formatter.into_source_text()
+            fixer.new_fix(
+                CompositeFix::Multiple(vec![
+                    Fix::delete(Span::new(
+                        static_mem_expr.object().span().end,
+                        static_mem_expr.span().end,
+                    )),
+                    Fix::new("toHaveLength", matcher_span),
+                ]),
+                Some(Cow::Owned(format!(
+                    "Use `toHaveLength` instead of `{}`.",
+                    matcher.name().unwrap()
+                ))),
+            )
+        });
```

Comment:
you should just be able to return a iterator (then colected into `RuleFix`

https://github.com/oxc-project/oxc/blob/34bfaf67922b5217584940ad49c1d363437fa9ee/crates/oxc_linter/src/rules/react/jsx_props_no_spread_multi.rs#L97-L104